subsys/bluetooth/mesh/net.c

```diff
@@ -266,14 +266,26 @@ bool bt_mesh_net_iv_update(uint32_t iv_index, bool iv_update)
 		}
 
 		if (iv_index == bt_mesh.iv_index + 1 && !iv_update) {
-			BT_WARN("Ignoring new index in normal mode");
-			return false;
-		}
+			if (!(IS_ENABLED(CONFIG_BT_MESH_IV_UPDATE_TEST) &&
+			      atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_TEST))) {
+				if (bt_mesh.ivu_duration < BT_MESH_IVU_MIN_HOURS) {
```

Comment:
Also, I'm wondering why are these lines added? I think only Line 268 to Line 271, were needed to be removed to enable IV index migration upon missed IV update.

Reply:
I guess it is because the requirement to follow 96 hours rule is mandatory.

Reply:
The requirement does not apply in the case as stated in the paragraph in the Jira ticket.

Reply:
Do we need to change this at all. The spec allow to choose  any of the 3 solutions, as it uses words as "may" and "or". The existing solution chose the last solution. This has been this way for several years. If we should change it it was not enough to do it as it said in the issue, because that would have given the same result with exit in the "!iv_upate" bellow.
The time (96 hours) was chosen as that is the only thing we have to go on. I will update with a  local variable.

Reply:
Yes, we still need a change, because if we don't do this there is no way for the device to recover by itself (and we don't provide any way that takes care of other options provided by "may") resulting in the device getting thrown out of the network and becoming unrechable.

Reply:
After discussion with @omkar3141, I disregarded the part about "The node makes the choice depending on the time since IV update and the likelihood that the node has missed the Secure Network beacons with the IV update Flag set to 1." as this does not specify any time. Also found that it is better to do a "IV Index Recovery procedure" than the "ignore the Secure Network beacon". This way the node will continue to function after a missed IV update. This also make this function the same way as if IV index is greater than IV index + 1.

Reply:
```
3.10.5 IV Update procedure
When a node is added to a network, the node is given an IV Index. If the node is added to a network when the network is in Normal operation, then it shall operate in Normal operation for at least 96 hours. If a node is added to a network while the network is in the IV Update in Progress state, then the node shall be given the new IV Index value and operate in Normal operation for at least 96 hours.
```

I don't see any reason why this rule should be ignored here just basing on the assumption the device was thrown out of the network. It doesn't seem correct.

Reply:
@alxelax  I don't understand how this rule is relevant here. The text you quoted is related to provisioning. In that text, the word "added" means "provisioned".

Reply:
@omkar3141  how did you determine here this 96 hours limitation is not acceptable here? Is there any attribute or flag that can guarantee 96 hours expired since the last IV Update procedure? Could you point it?

It seems the current implementation will align IV index of the whole network to the device with the highest IV number even device behaves incorrectly. The fact the devices have procedure time limitations will be ignored.

Even more, the specification requires quite strictly time limitation in one more place:
```
3.10.6 IV Index Recovery procedure
The node shall not execute more than one IV Index Recovery within a period of 192 hours.
```
That is not checked anywhere.  Probably it is out of this task since it has not been implemented yet.